File: cases/taint/go/dispatch-table-positive/dispatch_table_positive.go
package dataflowbench

func dfb_source() string { // DFB-SOURCE: dispatch-table-input
	return "tainted"
}

func dfb_sink(value string) {} // DFB-SINK: dispatch-table-sink

var table = map[string]func(value string){
	"leak": func(value string) { // DFB-WITNESS: dispatch-table-entry
		dfb_sink(value)
	},
	"drop": func(value string) {
		dfb_sink("clean")
	},
}

func run() {
	key := "leak"
	selected := table[key]
	selected(dfb_source())
}

File: cases/taint/csharp/dispatch-table-positive/DispatchTablePositive.cs
using System;
using System.Collections.Generic;

namespace DataFlowBench;

static class DispatchTablePositive
{
    static string dfb_source() // DFB-SOURCE: dispatch-table-input
    {
        return "tainted";
    }

    static void dfb_sink(string value) { } // DFB-SINK: dispatch-table-sink

    static void Run()
    {
        Dictionary<string, Action<string>> table = new Dictionary<string, Action<string>>();
        table["leak"] = value => // DFB-WITNESS: dispatch-table-entry
        {
            dfb_sink(value);
        };
        table["drop"] = value =>
        {
            dfb_sink("clean");
        };
        string key = "leak";
        Action<string> selected = table[key];
        selected(dfb_source());
    }
}
